_assets/js/carousel.js

```diff
@@ -0,0 +1,335 @@
+    liveregion.setAttribute('aria-atomic', 'true');
+    liveregion.setAttribute('class', 'liveregion hidden');
+    carousel.appendChild(liveregion);
+
+    // After the slide transitioned, remove the in-transition class, if focus should be set, set the tabindex attribute to -1 and focus the slide.
+    slides[0].parentNode.addEventListener('transitionend', function (event) {
+      let slide = event.target;
+      removeClass(slide, 'in-transition');
+      if (hasClass(slide, 'current')) {
+        if (setFocus) {
+          slide.setAttribute('tabindex', '-1');
+          slide.focus();
+          setFocus = false;
+        }
+      }
+    });
+
+    // When the mouse enters the carousel, suspend the animation.
+    carousel.addEventListener('mouseenter', suspendAnimation);
+
+    // When the mouse leaves the carousel, and the animation is suspended, start the animation.
+    carousel.addEventListener('mouseleave', function (event) {
+      if (animationSuspended) {
+        startAnimation();
+      }
+    });
+
+    // When the focus enters the carousel, suspend the animation
+    carousel.addEventListener('focusin', function (event) {
+      if (!hasClass(event.target, 'slide')) {
+        suspendAnimation();
+      }
+    });
+
+    // When the focus leaves the carousel, and the animation is suspended, start the animation
+    carousel.addEventListener('focusout', function (event) {
+      if (!hasClass(event.target, 'slide') && animationSuspended) {
+        startAnimation();
+      }
+    });
+
+    // Set the index (=current slide) to 0 – the first slide
+    index = 0;
+    setSlides(index);
+
+    // If the carousel is animated, advance to the
+    // next slide after 5s
+    if (settings.startAnimated) {
+      timer = setTimeout(nextSlide, 5000);
+    }
+  }
+
+  // Function to set a slide the current slide
+  function setSlides(new_current, setFocusHere, transition, announceItemHere) {
+    // Focus, transition and announce Item are optional parameters.
+    // focus denotes if the focus should be set after the
+    // carousel advanced to slide number new_current.
+    // transition denotes if the transition is going into the
+    // next or previous direction.
+    // If announceItem is set to true, the live region’s text is changed (and announced)
+    // Here defaults are set:
+
+    setFocus = typeof setFocusHere !== 'undefined' ? setFocusHere : false;
+    transition = typeof transition !== 'undefined' ? transition : 'none';
+    const announceItem = typeof announceItemHere !== 'undefined' ? announceItemHere : false;
+
+    new_current = parseFloat(new_current);
+
+    let length = slides.length;
+    let new_next = new_current + 1;
+    let new_prev = new_current - 1;
+
+    // If the next slide number is equal to the length,
+    // the next slide should be the first one of the slides.
+    // If the previous slide number is less than 0.
+    // the previous slide is the last of the slides.
```

Comment:
Suggest removing; this comment restates the code below it - [it doesn't explain why](https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/)

```suggestion
```